### PR TITLE
Fix fmod/remainder confusion in local planner

### PR DESCRIPTION
**Pull request by timonegk**
## Proposed changes
This might fix the problem of the robot sometimes turning in the wrong direction.

## Necessary checks
- [ ] Update package version
- [x] Run `catkin build`
- [ ] Write documentation
- [ ] Create issues for future work
- [ ] Test on your machine
- [ ] Test on the robot
- [x] Put the PR on our Project board

### CODE REVIEW

File: bitbots_local_planner/src/bitbots_local_planner.cpp [hunk 1]
```diff
@@ -84,23 +84,19 @@ namespace bitbots_local_planner
         tf2::fromMsg(current_pose_gm, current_pose);
 
         // Calculate the heading angle from our current position to the carrot
-        double walk_angle = std::fmod(
-            std::atan2(
-                goal_pose_.getOrigin().y() - current_pose.getOrigin().y(),
-                goal_pose_.getOrigin().x() - current_pose.getOrigin().x()),
-            2 * M_PI);
+        double walk_angle = std::atan2(
```
Comment (Flova) [inline bitbots_local_planner/src/bitbots_local_planner.cpp:87]: Afaik walk angle can be negative after this change. The interval before was [0, 2pi] and now it is [-pi, pi]. But I think we should be fine.

File: bitbots_local_planner/src/bitbots_local_planner.cpp [hunk 2]
```diff
@@ -84,23 +84,19 @@ namespace bitbots_local_planner
...
+            goal_pose_.getOrigin().y() - current_pose.getOrigin().y(),
+            goal_pose_.getOrigin().x() - current_pose.getOrigin().x());
 
         // Calculate the heading angle from our current position to the final position of the global plan
-        double final_walk_angle = std::fmod(
-            std::atan2(
-                end_pose_.getOrigin().y() - current_pose.getOrigin().y(),
-                end_pose_.getOrigin().x() - current_pose.getOrigin().x()),
-            2 * M_PI);
+        double final_walk_angle = std::atan2(
```
Comment (Flova) [inline bitbots_local_planner/src/bitbots_local_planner.cpp:92]: Same here

File: bitbots_local_planner/src/bitbots_local_planner.cpp [hunk 3]
```diff
@@ -84,23 +84,19 @@ namespace bitbots_local_planner
...
+            end_pose_.getOrigin().y() - current_pose.getOrigin().y(),
+            end_pose_.getOrigin().x() - current_pose.getOrigin().x());
 
         // Calculate the distance from our current position to the final position of the global plan
-        double distance = sqrt(
-            pow(end_pose_.getOrigin().y() - current_pose.getOrigin().y(), 2) +
-            pow(end_pose_.getOrigin().x() - current_pose.getOrigin().x(), 2));
+        double distance = std::hypot(
```
Comment (Flova) [inline bitbots_local_planner/src/bitbots_local_planner.cpp:97]: Cool

File: bitbots_local_planner/src/bitbots_local_planner.cpp [hunk 4]
```diff
@@ -119,12 +115,12 @@ namespace bitbots_local_planner
...
 
         // Get the min angle of the difference
-        double min_angle = (std::fmod(diff + M_PI, 2 * M_PI) - M_PI);
+        double min_angle = std::remainder(diff, 2 * M_PI);
         // Calculate our desired rotation velocity based on the angle difference and our max velocity
-        double vel = std::max(std::min(
-                                  config_.rotation_slow_down_factor * min_angle,
-                                  config_.max_rotation_vel),
-                              -config_.max_rotation_vel);
+        double vel = std::clamp(
```
Comment (Flova) [inline bitbots_local_planner/src/bitbots_local_planner.cpp:120]: Also cool

**Review by Flova — COMMENTED** [review]
I test this a bit more before approving it.

**Comment by Flova** [comment]
I tested this a bit and it works great so far.

**Review by SammyRamone — APPROVED** [review]

**Review by Flova — APPROVED** [review]

File: bitbots_local_planner/src/bitbots_local_planner.cpp (hunk 1 above)
Comment (timonegk) [reply bitbots_local_planner/src/bitbots_local_planner.cpp:87]: That is actually not the case, leaving `fmod` out does not change a thing. `fmod` guarantees the results to have the same sign as the input, and the output of `atan2` is in [-pi, pi]. So the output of `fmod` in this case is also in [-pi, pi], which is just the identity.

**Pull request merged**